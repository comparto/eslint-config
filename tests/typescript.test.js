const typescript = require('../src/rules/typescript')

test('typescript.js', () => {
  expect(typescript).toMatchSnapshot()
})
