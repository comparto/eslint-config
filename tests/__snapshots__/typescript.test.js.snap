// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript.js 1`] = `
Object {
  "overrides": Array [
    Object {
      "files": Array [
        "**/*.ts?(x)",
      ],
      "parser": "@typescript-eslint/parser",
      "plugins": Array [
        "@typescript-eslint",
      ],
      "rules": Object {
        "@typescript-eslint/adjacent-overload-signatures": "error",
        "@typescript-eslint/ban-ts-ignore": "error",
        "@typescript-eslint/ban-types": "error",
        "@typescript-eslint/camelcase": "error",
        "@typescript-eslint/class-name-casing": "error",
        "@typescript-eslint/consistent-type-assertions": "error",
        "@typescript-eslint/explicit-function-return-type": "warn",
        "@typescript-eslint/interface-name-prefix": "error",
        "@typescript-eslint/member-delimiter-style": "error",
        "@typescript-eslint/no-array-constructor": "error",
        "@typescript-eslint/no-empty-function": "error",
        "@typescript-eslint/no-empty-interface": "error",
        "@typescript-eslint/no-explicit-any": "warn",
        "@typescript-eslint/no-inferrable-types": "error",
        "@typescript-eslint/no-misused-new": "error",
        "@typescript-eslint/no-namespace": "error",
        "@typescript-eslint/no-non-null-assertion": "warn",
        "@typescript-eslint/no-this-alias": "error",
        "@typescript-eslint/no-unused-vars": "warn",
        "@typescript-eslint/no-use-before-define": "error",
        "@typescript-eslint/no-var-requires": "error",
        "@typescript-eslint/prefer-namespace-keyword": "error",
        "@typescript-eslint/triple-slash-reference": "error",
        "@typescript-eslint/type-annotation-spacing": "error",
        "camelcase": "off",
        "import/named": "off",
        "no-array-constructor": "off",
        "no-empty-function": "off",
        "no-unused-vars": "off",
        "no-use-before-define": "off",
        "no-var": "error",
        "prefer-const": "error",
        "prefer-rest-params": "error",
        "prefer-spread": "error",
      },
      "settings": Object {
        "import/external-module-folders": Array [
          "node_modules",
          "node_modules/@types",
        ],
        "import/parsers": Object {
          "@typescript-eslint/parser": Array [
            ".ts",
            ".tsx",
            ".d.ts",
          ],
        },
      },
    },
  ],
}
`;
